spec/jobs/create_github_repository_job_spec.rb

```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe AssignmentRepo::CreateGitHubRepositoryJob, type: :job do
+  include ActiveJob::TestHelper
+
+  subject { AssignmentRepo::CreateGitHubRepositoryJob }
+
+  let(:organization) { classroom_org }
+  let(:student)      { classroom_student }
+  let(:teacher)      { classroom_teacher }
+
+  let(:assignment) do
+    options = {
+      title: "Learn Elm",
+      starter_code_repo_id: 1_062_897,
+      organization: organization,
+      students_are_repo_admins: true
+    }
+
+    create(:assignment, options)
+  end
+
+  after do
+    clear_enqueued_jobs
+    clear_performed_jobs
+  end
+
+  describe "successful creation", :vcr do
+    after(:each) do
+      AssignmentRepo.destroy_all
+    end
+
+    it "uses the :create_repository queue" do
+      assert_performed_with(job: subject, args: [assignment, student], queue: "create_repository") do
+        subject.perform_later(assignment, student)
+      end
+    end
+
+    it "creates an AssignmentRepo as an outside_collaborator" do
+      subject.perform_now(assignment, student)
+
+      result = assignment.assignment_repos.first
+      expect(result.nil?).to be_falsy
+      expect(result.assignment).to eql(assignment)
+      expect(result.user).to eql(student)
```

Comment:
Style point on specs: Aim to have one `expect` per it block. If you need multiple `expects` split them up into separate it blocks and group them together logically in a `context`block.